### PR TITLE
Cluster autoscaler

### LINKED ISSUE #3370: Integrate cluster-autoscaler
**User Story**

As a user I want my clusters to autoscale to save cost

**Acceptance criteria**

* A fork of the RedHat cluster-api provider for CA is in the kubermatic org: https://github.com/kubernetes/autoscaler/pull/1866
* Prow has a postsubmit to build on commits of the `frobware:clusterapi-cloudprovider` branch
* The cluster-controller and openshift controller deploy the CA
* The usercluster controller deploys appropriate (Cluster-)RoleBindings for the CA to work
* The machine-controller puts an `cluster.k8s.io/machine: machine-ns/machine-name` annotation onto nodes

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Introduces support for autoscaling for Kubernetes > 1.13 or openshift, by adding the following annotations like this on the MachineDeployment:

```
 cluster.k8s.io/cluster-api-autoscaler-node-group-max-size: "10"
 cluster.k8s.io/cluster-api-autoscaler-node-group-min-size: "1"
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3370 

**Special notes for your reviewer**:

requires a new machine-controller release first.

/hold
**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/resources/clusterautoscaler/clusterautoscaler.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+				tag = "f1df2eb00ad9782b0c23184194008fc3068ad52c-1"
+			} else {
+				switch data.Cluster().Spec.Version.Minor() {
+				case 14:
+					tag = "fe5bee817ad9d37c8ce5e473af201c2f3fdf5b94-1"
+				}
+			}
+			if tag == "" {
+				return nil, fmt.Errorf("No matching autoscaler tag found for version %d", data.Cluster().Spec.Version.Minor())
+			}
```
Comment (mrIncompetent) [inline api/pkg/resources/clusterautoscaler/clusterautoscaler.go:31]: Move the tag logic into a own function + add a comment explaining why we do that

**Comment by mrIncompetent** [comment]
The autoscaler should not be deployed if the user did not explicitly opted it for this feature.
Otherwise we're wasting resources

**Comment by alvaroaleman** [comment]
> The autoscaler should not be deployed if the user did not explicitly opted it for this feature.
Otherwise we're wasting resources

I agree, we don't currently have any logic to determine that thought. Should we add an annotation for that for the time being?

**Comment by mrIncompetent** [comment]
> Should we add an annotation for that for the time being?

An annotation sounds good for the time being :+1: 
I'll create an issue to tackle cluster level feature flags

**Comment by alvaroaleman** [comment]
> I'll create an issue to tackle cluster level feature flags

IMHO that is part of https://github.com/kubermatic/kubermatic/issues/3568

**Comment by alvaroaleman** [comment]
@mrIncompetent The machine-controller update needed for this has merged, PTAL
/hold cancel

**Pull request merged**